litespeed-cache/thirdparty/lscwp-3rd-woocommerce.cls.php

```diff
@@ -962,5 +963,45 @@ public static function get_instance()
 
 		return self::$_instance ;
 	}
+
+	public static function pre_load()
+	{
+		// Auto purge for WooCommerce Advanced Bulk Edit plugin
+		// Bulk edit hook need to add to preload as it will die before detect
+		add_action( 'wp_ajax_wpmelon_adv_bulk_edit', 'LiteSpeed_Cache_ThirdParty_WooCommerce::bulk_edit_purge', 1 ) ;
+	}
+
+	public static function bulk_edit_purge()
+	{
+		$type = $_POST[ 'type' ] ;
+
+		if ( $type !== 'saveproducts' && empty( $_POST[ 'data' ] ) ) return ;
+
+		$config = LiteSpeed_Cache_API::config( 'wc_update_interval' ) ;
+		if ( is_null($config) ) {
+			$config = 0 ;
+		}
+
+		$stock_string_data = $_POST[ 'data' ] ;
+		$stock_string_arr = array() ;
+		foreach ( $stock_string_data as $stock_key => $stock_value ) {
+			$stock_string_arr = array_merge( $stock_string_arr, explode( '#^#', $stock_value ) ) ;
```

Comment:
Please add comments for the data structure.

Reply:
admin-ajax form data:

```action: wpmelon_adv_bulk_edit
type: saveproducts
nonce: afdc96369c
isfirst: 1
data[_regular_price]: 464$###0$###1#^#463$###0$###3#^#462$###0$###5#^#456$###0$###6#^#311$###0$###4
data[_sale_price]: 464$###0$###2#^#463$###0$###4#^#462$###0$###4
columns[_product_permalink]: 170
columns[product_type]: 100
columns[_visibility]: 190
columns[_stock_status]: 80
columns[_stock]: 80
columns[_tax_status]: 80
columns[_featured]: 80
columns[post_status]: 70
columns[post_content]: 250
columns[_sku]: 60
columns[_sale_price]: 80
columns[_regular_price]: 80
columns[product_cat]: 150
columns[_product_image_gallery]: 90
columns[_thumbnail_id]: 80
columns[post_title]: 250
columns[ID]: 60
columns[sel]: 30
filters: 
```

Long version here, will also add the major part on the code.